Frameworks/CoreText/CTFont.mm

```diff
@@ -259,124 +315,95 @@ CFDictionaryRef CTFontCopyTraits(CTFontRef font) {
 }
 
 /**
- @Status Caveat
- @Notes Always returns font family name
+ @Status Interoperable
 */
 CFStringRef CTFontCopyPostScriptName(CTFontRef font) {
-    return (CFStringRef)[[(UIFont*)font fontName] retain];
+    return CTFontCopyName(font, kCTFontPostScriptNameKey);
 }
 
 /**
  @Status Interoperable
 */
 CFStringRef CTFontCopyFamilyName(CTFontRef font) {
-    return (CFStringRef)[[(UIFont*)font fontName] retain];
+    return CTFontCopyName(font, kCTFontFamilyNameKey);
 }
 
 /**
- @Status Caveat
- @Notes Calls CTFontCopyName which supports limited encodings.
+ @Status Interoperable
 */
 CFStringRef CTFontCopyFullName(CTFontRef font) {
     return CTFontCopyName(font, kCTFontFullNameKey);
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 CFStringRef CTFontCopyDisplayName(CTFontRef font) {
-    UNIMPLEMENTED();
-    return StubReturn();
+    // This should work okay as a display name
+    return CTFontCopyName(font, kCTFontFullNameKey);
 }
 
 /**
- @Status Caveat
- @Notes Supports only English language and limited encodings.
+ @Status Interoperable
 */
 CFStringRef CTFontCopyName(CTFontRef font, CFStringRef nameKey) {
     if (nameKey == nullptr || font == nullptr) {
         return nullptr;
     }
 
-    //  Get the font
-    FT_Face face = reinterpret_cast<FT_Face>([(UIFont*)font _sizingFontHandle]);
-
-    // Style name is not present in the name table associated with the font.
-    // It is stored in FT_Face.
-    if (nameKey == kCTFontStyleNameKey) {
-        return static_cast<CFStringRef>([[NSString stringWithCString:face->style_name] retain]);
+    ComPtr<IDWriteLocalizedStrings> name;
+    BOOL exists;
+    DWRITE_INFORMATIONAL_STRING_ID informationalStringId;
+
+    if (CFEqual(nameKey, kCTFontCopyrightNameKey)) {
```

Comment:
probably, same reason

---
In reply to: [78227559](https://github.com/Microsoft/WinObjC/pull/987#discussion_r78227559) [](ancestors = 78227559)